### PR TITLE
MM-18786: Removed UNSAFE methods from setting components

### PR DESCRIPTION
**Pull request by CSBatchelor**
#### Summary

Removes the deprecated lifecycle methods from settings components.

setting_picture:
-  Moved state changes to getDerivedStateFromProps
- Added file to state so we can check for changes
-  Moved addditional code to componentDidUpdate

multiselect_list:
- Removed toSelect after debugging because it didn't seem necessary.
- toSelect also added some complications to refactoring this.
- Moved additional code to componentDidUpdate

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/12339

### CODE REVIEW

File: components/setting_picture.jsx [hunk 1]
```diff
@@ -38,6 +38,13 @@ export default class SettingPicture extends Component {
         maxFileSize: PropTypes.number,
     };
 
+    static getDerivedStateFromProps(props, state) {
```
Comment (saturninoabril) [inline components/setting_picture.jsx]: Looks like this is not necessary.  Could you please check?

File: components/setting_picture.jsx [hunk 2]
```diff
@@ -46,6 +53,7 @@ export default class SettingPicture extends Component {
         this.confirmButton = React.createRef();
 
         this.state = {
+            file: null,
```
Comment (saturninoabril) [inline components/setting_picture.jsx]: Same here.

File: components/setting_picture.jsx (hunk 1 above)
Comment (CSBatchelor) [reply components/setting_picture.jsx]: Confirmed and removed in the next commit.

**Comment by CSBatchelor** [comment]
@saturninoabril Thanks! New commit submitted to address your comments.

**Comment by deanwhillier** [comment]
Thanks so much @CSBatchelor! Would you mind merging master into your branch and pushing when you have a minute so we can complete the merge?

**Comment by CSBatchelor** [comment]
No problem, it's been a great experience.

**Pull request merged**